keras_nlp/utils/preset_utils.py

```diff
@@ -382,9 +458,14 @@ def upload_preset(
                 "'hf://username/bert_base_en' or 'hf://bert_case_en' to implicitly"
                 f"upload to your user account. Received: URI={uri}."
             ) from e
-        huggingface_hub.upload_folder(
-            repo_id=repo_url.repo_id, folder_path=preset
-        )
+        try:
+            huggingface_hub.upload_folder(
+                repo_id=repo_url.repo_id, folder_path=preset
+            )
+        finally:
+            # Clean up the preset directory in case user attempts to upload the
+            # preset directory into Kaggle hub as well.
+            delete_model_card(preset)
```

Comment:
Sorry, last thing I forgot to mention. If the user push a model to an existing repo, we don't want to generate a model card as it might overwrite content that the user has manually written already. So I would implement a logic like that:
1. create the repo with `exists_ok=True`
2. check if `file_exists`
3. generate model card (if doesn't exist)
4. upload folder
5. delete local model card (if generated in 3.)

```suggestion
        has_model_card = huggingface_hub.file_exists(repo_id=repo_url.repo_id, filename=README_FILE)
        if not has_model_card:
            # Remote repo do not contain a model card => create one
            create_model_card(preset)
        try:
            huggingface_hub.upload_folder(
                repo_id=repo_url.repo_id, folder_path=preset
            )
        finally:
            if not requires_model_card:
                # Clean up the preset directory in case user attempts to upload the
                # preset directory into Kaggle hub as well.
                delete_model_card(preset)
```

(and do not forget to remove the `create_model_card` call above L449)